pywbem_mock/_wbemconnection_mock.py

```diff
@@ -588,7 +590,9 @@ def compile_mof_string(self, mof_str, namespace=None, search_paths=None,
             The name of the target CIM namespace in the CIM repository. This
             namespace is also used for lookup of any existing or dependent
             CIM objects. If `None`, the default namespace of the connection is
-            used.
+            used. A namespace defined in a namespace pragma of the MOF
+            has precedence over the namespace parameter.  The namespace
+            specified in this parameter or the MOF itself must exist.
```

Comment:
see above on "precedence".

Reply:
Cleaned up

Reply:
I don't see a change?

Reply:
removed use of word precedence and change to note that the pragma becomes the namespace from the point of declaration in the compile element (string/file).  Same statement used in all of the 4 references to namespace parameter in compiling mof.

Reply:
It is now much better but it does not yet cover the behavior that at the end of a MOF file that had a pragma namespace, the namespace before that MOF file gets used again.

Reply:
We looked into the code and convinced ourselves that we were wrong with the assumption the compiler would implement a nesting of namespaces along the file boundaries. Instead, the compiler only maintains one globally set target namespace that is changed when encountering a pragma namespace directive, but is not reset when encountering the end of file. Given that, the current text is correct.